### PR TITLE
filter out maintenance in VisitTimeDistribution

### PR DESCRIPTION
**Pull request by JaerongA**
subject to change (WIP)

### CODE REVIEW

File: aeon/dj_pipeline/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+def _make_maintenance_filter(position_df, maintenance_filter, start, stop):
+    """Make a boolean filter for eliminating maintenance period"""
+
+    maintenance_filter += (position_df.index >= start) & (position_df.index <= stop)
+    return maintenance_filter
```
Comment (ttngu207) [inline aeon/dj_pipeline/utils/__init__.py:5]: This is pretty much a one-liner code, I'd suggest not having this refactored as a separate `_make_maintenance_filter` function, but just copy that one line directly where you have it used.
I'd argue that it is more readable than having this refactored function.

Comment (JaerongA) [reply aeon/dj_pipeline/utils/__init__.py]: Yes, that makes sense. Removed that one.

File: aeon/dj_pipeline/analysis/visit_analysis.py [hunk 2]
```diff
@@ -292,6 +308,41 @@ def make(self, key):
...
+
+            end_timestamps = np.array(
+                np.append(
+                    daily_log_df.loc[
+                        daily_log_df["message"] == "Experiment", "message_time"
+                    ],
+                    day_end,
+                ),
+                dtype="datetime64[ns]",
+            )  # end time
```
Comment (lochhh) [inline aeon/dj_pipeline/analysis/visit_analysis.py:336]:
```suggestion
            start_timestamps = daily_log_df.loc[
                daily_log_df["message"] == "Maintenance", "message_time"
            ]  # maintenance start time

            end_timestamps = daily_log_df.loc[
                daily_log_df["message"] == "Experiment", "message_time"
            ]  # maintenance end time
```

File: aeon/dj_pipeline/analysis/visit_analysis.py [hunk 3]
```diff
@@ -464,6 +529,41 @@ def make(self, key):
...
```
Comment (lochhh) [inline aeon/dj_pipeline/analysis/visit_analysis.py:557]:
```suggestion
            start_timestamps = daily_log_df.loc[
                daily_log_df["message"] == "Maintenance", "message_time"
            ]  # maintenance start time

            end_timestamps = daily_log_df.loc[
                daily_log_df["message"] == "Experiment", "message_time"
            ]  # maintenance end time
```

**Review by lochhh — COMMENTED** [review]
Removing ``day_start`` and ``day_end`` from maintenance timestamps as we want to keep the data between 1) ``day_start`` and the first maintenance (of the day) and 2) the last maintenance (of the day) and ``day_end``

**Comment by lochhh** [comment]
@JaerongA Not sure whether I mentioned this but the final 'maintenance' timestamp (after removing duplicates) marks the end of the visit, i.e. we also discard what's between that 'maintenance' timestamp and ``visit_end``

**Comment by JaerongA** [comment]
> @JaerongA Not sure whether I mentioned this but the final 'maintenance' timestamp (after removing duplicates) marks the end of the visit, i.e. we also discard what's between that 'maintenance' timestamp and `visit_end`

Yes, that makes sense. I'll restructure the code and fix things to reflect the comments during our meeting.

**Pull request closed without merging**